docs/apm.asciidoc

```diff
@@ -0,0 +1,234 @@
++
+[source,yaml]
+----
+apiVersion: apm.k8s.elastic.co/v1alpha1
+kind: ApmServer
+metadata:
+  name: apm-server-quickstart
+  namespace: default
+spec:
+  version: "7.2.0"
+  nodeCount: 1
+  secureSettings:
+    secretName: apm-secret-settings
+  config:
+    output:
+      elasticsearch:
+        hosts: ["my-own-elasticsearch-cluster:9200"]
+        username: elastic
+        password: "${ES_PASSWORD}"
+        protocol: "https"
+        ssl.certificate_authorities: ["/usr/share/apm-server/config/elasticsearch-ca/tls.crt"]
+  podTemplate:
+    spec:
+      containers:
+      - name: apm-server
+        volumeMounts:
+        - mountPath: /usr/share/apm-server/config/elasticsearch-ca
+          name: elasticsearch-ca
+          readOnly: true
+      volumes:
+      - name: elasticsearch-ca
+        secret:
+          defaultMode: 420
+          optional: false
+          secretName: es-ca # This is the secret that holds the Elasticsearch CA cert
+----
+
+[float]
+[id="{p}-apm-tls"]
+==== TLS Certificates
+
+By default the operator manages a private CA and generates a self-signed certificate used to secure the communication between APM agents and the server.
+
+This behavior and the relevant configuration is identical to what is done for Elasticsearch and Kibana. You can refer to link:k8s-accessing-elastic-services.html#k8s-setting-up-your-own-certificate[this documentation] to understand how to use your own certificate to configure the TLS endpoint of the APM Server.
+
+You can refer to the https://www.elastic.co/guide/en/apm/agent/index.html[APM agents documentation] for further details about how to configure them to work with custom certificates.
+
+[float]
+[id="{p}-apm-connecting"]
+=== Connecting to the APM Server
+[float]
+[id="{p}-apm-service"]
+==== APM Server service
+
+The APM Server is exposed with a Service. You can refer to link:k8s-accessing-elastic-services.html[this documentation] for information on accessing it.
+
+You can retrieve the list of all the APM Services with following command:
+
+[source,sh]
+----
+kubectl get service --selector='common.k8s.elastic.co/type=apm-server'
+----
+
+[source,sh]
+----
+NAME                             TYPE        CLUSTER-IP   EXTERNAL-IP   PORT(S)    AGE
+apm-server-quickstart-apm-http   ClusterIP   10.0.1.252   <none>        8200/TCP   154m
+----
+[float]
+[id="{p}-apm-secret-token"]
+==== APM Server secret token
+
+The operator generates an authorization token that agents must send to authenticate themselves to the APM Server.
+
+This token is stored in a secret named `{APM-server-name}-apm-token` and can be retrieved with the following command:
+
+[source,sh]
+----
+kubectl get secret/apm-server-quickstart-apm-token  -o go-template='{{index .data "secret-token" | base64decode}}'
+----
+
+For any additional information about the APM Server please refer to https://www.elastic.co/guide/en/apm/server/current/index.html[this documentation].
```

Comment:
For more information about the APM Server, see https://www.elastic.co/guide/en/apm/server/current/index.html[APM Server Reference].